### PR TITLE
feat(database): support row-level selection

### PR DESCRIPTION
**Pull request by zqran**
close: #2468 
alo close: #2467 

https://github.com/toeverything/blocksuite/assets/15389209/0737e3d8-ccf8-43df-baf0-ea4d4cf202dc

### CODE REVIEW

File: packages/blocks/src/page-block/default/selection-manager/database-selection-manager/utils.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+function clearDatabaseRowsSelection(database: HTMLElement) {
+  const allRows = database.querySelectorAll<HTMLElement>('.database-row');
+  allRows.forEach(row => {
+    row.classList.remove(DATABASE_ROW_SELECTED_CLASS);
+  });
+}
+
+function getClosestRowId(element: Element): number {
+  const rowId = element.closest('.database-row')?.getAttribute('data-row-id');
+  assertExists(rowId);
```
Comment (fundon) [inline packages/blocks/src/page-block/default/selection-manager/database-selection-manager/utils.ts:70]: Screenshot 2023-05-15 at 1 09 12 PM


Throws an error when moving to header.

Comment (zqran) [reply packages/blocks/src/page-block/default/selection-manager/database-selection-manager/utils.ts]: fixed

**Review by fundon — APPROVED** [review]

File: packages/blocks/src/page-block/default/default-page-block.ts [hunk 2]
```diff
@@ -447,6 +455,18 @@ export class DefaultPageBlockComponent
     slots.embedEditingStateUpdated.on(embedEditingState => {
       this._embedEditingState = embedEditingState;
     });
+    slots.databaseTableUpdated.on(state => {
```
Comment (Saul-Mirone) [inline packages/blocks/src/page-block/default/default-page-block.ts:458]: Can you move this slot into database model?

File: packages/blocks/src/page-block/default/utils.ts [hunk 3]
```diff
@@ -539,6 +539,24 @@ export function createDragHandle(pageBlock: DefaultPageBlockComponent) {
       pageBlock.selection.state.type = dragging ? 'block:drag' : 'block';
     },
     setSelectedBlock(modelState: EditingState | null) {
+      if (modelState) {
```
Comment (Saul-Mirone) [inline packages/blocks/src/page-block/default/utils.ts:542]: Instead of handle database selection here, maybe we can move this into database service.

**Review by Saul-Mirone — APPROVED** [review]

**Pull request merged**